components/public-api/typescript-common/src/public-api-converter.ts

```diff
@@ -1211,6 +1213,47 @@ export class PublicAPIConverter {
                 );
             }
         }
+
+        const capitalize = (input: string) => {
+            return input.charAt(0).toUpperCase() + input.slice(1);
+        }
+
+        // This is a hack mimicking the supervisor behavior of adding dynamic IDE tasks https://github.com/gitpod-io/gitpod/blob/e7d79c355e2cd6ac34056ea52d7bdcda45975839/components/ide-service/pkg/server/server.go#L508-L540
+        const jetbrainsIdes = prebuild.workspace.config.jetbrains;
+        if (jetbrainsIdes) {
+            for (const ide in jetbrainsIdes) {
```

Comment:
ide maybe any random string like `not_exists`, we need to check it's a available option first

Reply:
I thought we do some parsing here? The types seem to only contain the supported JB IDEs

Reply:
We do it the same in ide-service. Add a guard should be safer

Reply:
It seems we handle this already before even starting the prebuild with `InvalidGitpodYMLError`:
```
Invalid gitpod.yml: should NOT have additional properties
```

Reply:
You're right, I tried with incorrect .gitpod.yml. Prebuild is not triggered, but for me as a user it's weird: I'm not aware my prebuilds is not triggered because of incorrect formatting of .gitpod.yml although I have pushed lots of commits already

Reply:
I quite like how @dependabot does it, maybe we could look into it as well in the future.

<img width="1031" alt="image" src="https://github.com/gitpod-io/gitpod/assets/29888641/d410c148-8803-4a7a-ae31-d987e97a89ae">